### ISSUE TITLE
`scalafmt-dynamic` uses unsupported library `coursier-small`

### ISSUE THREAD

**Issue by mihaisoloi** [scalameta/scalafmt#1507]
- **.scalafmt.conf Version**: 2.0.1
- **sbt-plugin Version** 2.0.5

## Steps

Remove library dependency from `build.sbt` and replace with proper coursier dependency if necessary:
```scala
"com.geirsson" %% "coursier-small" % "1.3.1",
```

## Problem

Scalafmt uses the unsupported library https://github.com/olafurpg/coursier-small which hard-codes repositories, totally ignoring the `COURSIER_REPOSITORIES` environment variable. In setups where people use maven proxies to cache dependencies and not hit the main repo + have the round-trip performed, this breaks `scalafmt` by not being able to get the proper version:

```
[error] failed to resolve Scalafmt version '2.0.1': /src/.scalafmt.conf
[error] (Compile / scalafmtCheck) failed to resolve Scalafmt version '2.0.1': /src/.scalafmt.conf
```

## Expectation

I would like the `scalafmt-dynamic` to use the coursier API directly https://get-coursier.io/docs/api

**Comment by mihaisoloi** [comment]
Duplicate of https://github.com/scalameta/scalafmt/pull/1443

**Comment by olafurpg** [comment]
Thank you for reporting! This is fixed in #1511 and released in 2.1.0-RC2
